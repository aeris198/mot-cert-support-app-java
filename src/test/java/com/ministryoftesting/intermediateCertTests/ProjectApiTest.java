package com.ministryoftesting.intermediateCertTests;

import com.ministryoftesting.api.TimesheetManagerApplication;
import com.ministryoftesting.models.Credentials;
import io.restassured.response.Response;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import static io.restassured.RestAssured.given;

@ExtendWith(SpringExtension.class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT, classes = TimesheetManagerApplication.class)
@ActiveProfiles("dev")
public class ProjectApiTest {

    @Test
    public void testGettingProject2() {
        Credentials credentials = given()
                .body("{\"email\":\"admin@test.com\",\"password\":\"password123\"}")
                .contentType("application/json")
                .post("/v1/auth/login")
                .as(Credentials.class);

        String token = credentials.getToken();

        Response response = given()
                .header("Authorization", "Bearer " + token)
                .get("/v1/project/2");

        Approvals.verify(response.getBody().prettyPrint());
    }

}
