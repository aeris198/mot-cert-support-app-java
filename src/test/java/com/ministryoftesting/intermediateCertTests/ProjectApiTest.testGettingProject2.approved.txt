{
    "name": "Project 2",
    "description": "This is the second project",
    "total": 0,
    "entries": [
        
    ]
}